syntax = "proto3";

package stress;

// simplest message that can be made arbitrarily large
message GetMessage {
    string key = 1;
}

// simple message that can be made arbitrarily large
// by making the strings arbitrarily large
message PutMessage {
    string key = 1;
    string value = 2;
}

// two leaves
message Msg1L {
    GetMessage left = 1;
    GetMessage right = 2;
}

// triple nested (four leaves)
message Msg2L {
    Msg1L left = 1;
    Msg1L right = 2;
}

// eight leaves
message Msg3L {
    Msg2L left = 1;
    Msg2L right = 2; 
}

// 16 leaves
message Msg4L {
    Msg3L left = 1;
    Msg3L right = 2;
}

// 32 leaves
message Msg5L {
    Msg4L left = 1;
    Msg4L right = 2;
}
